src/frameworks/next/utils.ts

```diff
@@ -0,0 +1,63 @@
+import type { Header, Redirect, Rewrite } from "next/dist/lib/load-custom-routes";
+import { isUrl } from "../utils";
+
+/**
+ * Whether the given path has a regex or not.
+ * According to the Next.js documentation:
+ * ```md
+ *  To match a regex path you can wrap the regex in parentheses
+ *  after a parameter, for example /post/:slug(\\d{1,}) will match /post/123
+ *  but not /post/abc.
+ * ```
+ * See: https://nextjs.org/docs/api-reference/next.config.js/redirects#regex-path-matching
+ */
+export function pathHasRegex(path: string): boolean {
+  // finds parentheses that are not preceded by double backslashes
+  const regex = /(?<!\\)\(/;
```

Comment:
I don't think it would hurt readability to make this a one liner, e.g, `return /(?<!\\)\(/.test(path)`